ts/live-compositor/src/components/Image.ts

```diff
@@ -1,21 +1,105 @@
+import { createElement, useContext, useEffect, useState } from 'react';
 import type { ComponentBaseProps, SceneComponent } from '../component.js';
 import { createCompositorComponent } from '../component.js';
-import type { Api } from '../index.js';
-
-export type ImageProps = Omit<ComponentBaseProps, 'children'> & {
-  /**
-   * Id of a component.
-   */
-  id?: Api.ComponentId;
-  /**
-   * Id of an image. It identifies an image registered using `LiveCompositor.registerImage`.
-   */
-  imageId: Api.RendererId;
-};
-
-const Image = createCompositorComponent<ImageProps>(sceneBuilder);
-
-function sceneBuilder(props: ImageProps, _children: SceneComponent[]): Api.Component {
+import { View, type Api } from '../index.js';
+import { imageRefIntoRawId } from '../types/refs/imageRef.js';
+import { newInternalImageId } from '../context/internalImageIdManager.js';
+import { newBlockingTask } from '../hooks.js';
+import { LiveCompositorContext } from '../context/index.js';
+import { isValidImageType } from '../types/utils.js';
+
+export type ImageProps = Omit<ComponentBaseProps, 'children'> &
+  (
+    | {
+        imageId: Api.RendererId;
+        source?: never; // Ensuring 'source' cannot be used alongside 'imageId'
+      }
+    | {
+        source: string;
+        imageId?: never; // Ensuring 'imageId' cannot be used alongside 'source'
+      }
+  );
+
+type ImageSceneBuliderProps = Omit<ImageProps, 'imageId'> & { imageId: string };
+
+export const InnerImage = createCompositorComponent<ImageSceneBuliderProps>(sceneBuilder);
+
+function Image(props: ImageProps) {
+  const ctx = useContext(LiveCompositorContext);
+  const [imageId, setImageId] = useState(0);
```

Comment:
maybe internalImageId or autoImageId because it is confusing to distinguish this id from props one